README.md

```diff
@@ -1,107 +1,149 @@
-`ocamlfind` and you just need to add `bap` to your project. For
-example, if you use `oasis`, then you should add `bap` to the
-`BuildDepends` field. If you are using `ocamlbuild` with the
-`ocamlfind` plugin, then you should add `package(bap)` or `pkg_bap` to
-your `_tags` file.
+## Python
 
+OK, if the previous example doesn't make any sense for you, then you
+can try our
+[Python bindings](https://github.com/BinaryAnalysisPlatform/bap-python).
+Install them with `pip install bap` (you still need to install `bap`
+before hand). Here is the same example, but in Python:
 
-## Using from top-level
+```python
+import bap
+from bap.adt import Visitor
 
-It maybe a good idea to learn how to use our library by playing in an
-OCaml top-level. If you have installed `utop`, then you can just use
-our `baptop` script to run `utop` with `bap` extensions:
+class Counter(Visitor) :
+    def __init__(self):
+        self.jmps = 0
+        self.total = 0
+
+    def enter_Jmp(self,jmp):
+        self.jmps += 1
+
+    def enter_Term(self,t):
+        self.total += 1
+
+proj = bap.run('/bin/true')
+count = Counter()
+count.run(proj.program)
+print("ratio = {0}/{1} = {2}".format(count.jmps, count.total,
+                                     count.jmps/float(count.total)))
+```
+
+
+## baptop
+
+BAP also ships an interactive toplevel, aka REPL. This is a shell-like
+program, that will interactively evaluate OCaml instructions, and
+print results. Just run:
 
 ```bash
 $ baptop
 ```
 
-Now, you can play with BAP. The following example, will create a
-project from
-[coreutils_O2_true](https://github.com/BinaryAnalysisPlatform/arm-binaries/raw/master/coreutils/coreutils_O2_true)
-file, build callgraph of a program, control flow graph and dominance
-tree of a `main` function.
+Now, you can play with BAP. The following example will open a file,
+build callgraph of a program, and a control flow graph and dominance
+tree of a function.
 
 ```ocaml
-utop # open Core_kernel.Std;;
-utop # open Bap.Std;;
-utop # let proj = Project.from_file "coreutils_O2_true" |> ok_exn;;
-utop # let prog = Project.program proj;;
-utop # let cg = Program.to_graph prog;;
-utop # let main = Term.find_exn sub_t prog Tid.(!"@main");;
-utop # let cfg = Sub.to_cfg main;;
-utop # module G = Graphlib.Ir;;
-utop # let entry = Option.value_exn (Term.first blk_t main);;
-utop # let dom_tree = Graphlib.dominators (module G) cfg (G.Node.create entry);;
+open Core_kernel.Std;;
+open Bap.Std;;
+open Graphlib.Std;;
+let rooter = Rooter.Factory.find "byteweight" |> Option.value_exn;;
+let proj = Project.create ~rooter (Project.Input.file "/bin/true") |> ok_exn;;
+let prog = Project.program proj;;
+let cg = Program.to_graph prog;;
+let sub = Term.first sub_t prog |> Option.value_exn;;
+let cfg = Sub.to_cfg sub;;
+module G = Graphs.Ir;;
+let entry = Option.value_exn (Term.first blk_t sub);;
+let dom_tree = Graphlib.dominators (module G) cfg (G.Node.create entry);;
 ```
```

Comment:
Nothing will have printed here since everything was bound to variables. Maybe query a variable to make it print something?